Library/Homebrew/utils/github.rb

```diff
@@ -337,6 +337,43 @@ def print_pull_requests_matching(query)
     prs.each { |i| puts "#{i["title"]} (#{i["html_url"]})" }
   end
 
+  def fetch_pull_requests(query, tap_full_name, state: nil)
+    issues_for_formula(query, tap_full_name: tap_full_name, state: state).select do |pr|
+      pr["html_url"].include?("/pull/") &&
+        /(^|\s)#{Regexp.quote(query)}(:|\s|$)/i =~ pr["title"]
+    end
+  rescue GitHub::RateLimitExceededError => e
+    opoo e.message
+    []
+  end
+
+  def check_for_duplicate_pull_requests(formula, tap_full_name, version, args, fetch_pr = false)
+    # check for open requests
+    pull_requests = fetch_pull_requests(formula.name, tap_full_name, state: "open")
+
+    # if we haven't already found open requests, try for an exact match across all requests
+    pull_requests = fetch_pull_requests("#{formula.name} #{version}", tap_full_name) if pull_requests.blank?
+    return if pull_requests.blank?
+
+    return pull_requests.map { |pr| { title: pr["title"], url: pr["html_url"] } } if fetch_pr
+
+    duplicates_message = <<~EOS
+      These pull requests may be duplicates:
+      #{pull_requests.map { |pr| "#{pr["title"]} #{pr["html_url"]}" }.join("\n")}
+    EOS
+    error_message = "Duplicate PRs should not be opened. Use --force to override this error."
+    if args.force? && !args.quiet?
+      opoo duplicates_message
+    elsif !args.force? && args.quiet?
+      odie error_message
+    elsif !args.force?
+      odie <<~EOS
```

Comment:
Are you sure you want this to `odie`and exit here?

Reply:
@MikeMcQuaid yes please, that functionality is for `bump-formula-pr` command which kills the script and outputs a duplicate error message.

Reply:
This doesn't seem desirable for `brew bump`, though? I'd suggest that this method removes the `fetch_pr` arguments, always returns `pull_requests.map ...` above and you move the `duplicates_message` and below logic back into `bump-formula-pr`